test/tests/bidi.json

```diff
@@ -114,7 +115,7 @@
     {
         "description":  " name... excludes U+FFFD and U+061C -- this pases as name -> [bidi] name-start *name-char",
         "src": ".local $\u061Cfoo = {1} {{ {$\u061Cfoo} }}",
-        "exp": " 1 "
+        "exp": " \u20681\u2069 "
```

Comment:
I guess in this case, the direction of the message is determined to be `unknown`? Which is a little surprising to me, since the only bidi control character appears in one of the variable names, which I would expect not to have an effect on the formatted result of the message. Maybe what's throwing me off is that the explanation of the "Default Bidi Strategy" at the end of [this section](https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#handling-bidirectional-text) doesn't say how the message directionality is determined?

Reply:
The bidi controls in the syntax source don't matter here at all, and the message has LTR direction because of the `en-US` locale. It's the placeholder's direction that is unknown. For it to be known, we'd need to introspect the `1` string with something like [UAX#9 P2](https://unicode.org/reports/tr9/#P2), which will ultimately be done by the rendering engine, and even if we do, the direction would still be unknown because European numbers are only weakly directional. And so this is appropriately reflected by FSI/PDI isolating it.

This is much the same thing as I mention in https://github.com/unicode-org/message-format-wg/pull/917#discussion_r1818124771, but for LTR instead of RTL content.

> Maybe what's throwing me off is that the explanation of the "Default Bidi Strategy" at the end of [this section](https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#handling-bidirectional-text) doesn't say how the message directionality is determined?

Good point. We may want to add a mention under [Formatting Context](https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#formatting-context) about deducing the message direction from the locale if it's not explicitly given.

Reply:
If not explicitly provided, the direction of the message should be computed from the locale. As @eemeli notes, controls in the source don't matter. As noted elsewhere, you can't just look at the rendered placeholder to determine its desired direction--even strongly directional characters do not necessarily provide an accurate signal.